src/m365/aad/commands/app/app-set.ts

```diff
@@ -206,13 +214,94 @@ class AadAppSetCommand extends GraphCommand {
       .then(_ => Promise.resolve(objectId));
   }
 
+  private configureCertificate(args: CommandArgs, objectId: string, logger: Logger): Promise<void> {
+    if (!args.options.certificateFile && !args.options.certificateBase64Encoded) {
+      return Promise.resolve();
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Setting certificate for Azure AD app...`);
+    }
+
+    return this.getCertificateBase64Encoded(args, logger).then((certificateBase64Encoded) => {
+      const getAppRequestOptions: AxiosRequestConfig = {
+        url: `${this.resource}/v1.0/myorganization/applications/${objectId}?$select=keyCredentials`,
+        headers: {
+          'content-type': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      return request
+        .get<Application>(getAppRequestOptions)
+        .then((application: Application): Promise<void> => {
+          
+          const keyCredentials = application.keyCredentials as KeyCredential[];
+          
+          // The graph types of  defines the 'key' property of KeyCredential as 'NullableOption<number>'
+          // while it is a base64 encoded string. This is any is used here.
+          if (keyCredentials.every(existingCredential => existingCredential.key !== certificateBase64Encoded as any)) {
+            const newKeyCredential = {
+              type: "AsymmetricX509Cert",          
+              usage: "Verify",
+              displayName: args.options.certificateDisplayName,
+              key: certificateBase64Encoded
+            } as any;
+
+            keyCredentials.push(newKeyCredential);
+               
+            const requestOptions: AxiosRequestConfig = {
+              url: `${this.resource}/v1.0/myorganization/applications/${objectId}`,
+              headers: {
+                'content-type': 'application/json;odata.metadata=none'
+              },
+              responseType: 'json',
+              data: {
+                keyCredentials: keyCredentials
+              }
+            };
+
+            return request.patch(requestOptions);
+          }
+
+          return Promise.resolve();
+        })
+        .then(_ => Promise.resolve());
+    });
+  }
+
+  private getCertificateBase64Encoded(args: CommandArgs, logger: Logger): Promise<string> {
+    if (args.options.certificateBase64Encoded) {
+      return Promise.resolve(args.options.certificateBase64Encoded);
+    }
+    
+    if (fs.existsSync(args.options.certificateFile as string)) {
```

Comment:
We can simplify this code by moving this logic to the `validate` method and here assume that the file exists.